pkg/driver/controller.go

```diff
@@ -0,0 +1,359 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package driver
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/google/uuid"
+	"github.com/kubernetes-sigs/aws-efs-csi-driver/pkg/cloud"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog"
+)
+
+const (
+	AccessPointMode     = "efs-ap"
+	FsId                = "fileSystemId"
+	GidMin              = "gidRangeStart"
+	GidMax              = "gidRangeEnd"
+	DirectoryPerms      = "directoryPerms"
+	BasePath            = "basePath"
+	ProvisioningMode    = "provisioningMode"
+	DefaultGidMin       = 50000
+	DefaultGidMax       = 7000000
```

Comment:
isn't the max 65534 ? (depends on Os maybe?)

Any docs for the new parameters or example storageclass? Looksl ike ProvisioningMode is required. As for GID/direcotryPerms, I guess most will leave the GID parameters default given that it appears the GID has no effect on permissions for access points within the cluster as we discussed below (any Pod that mounts the access point will have access no matter their GID).

Reply:
Added this number based on what EFS supports. More info the default max supported by EFS here - https://docs.aws.amazon.com/efs/latest/ug/create-access-point.html